src/DistributedLocking/ILockProvider.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Cascade\DistributedLocking;
+
+interface ILockProvider
```

Comment:
Most our PHP is `FooInterface` rather than `IFoo`. Change of opinion? Think we did it, because that's generally how sensio does it. But can't remember exactly, and I'm cool with this way as well.